react-custom-hooks/client/src/components/useTodos.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+
+const url = (path) => `${process.env.REACT_APP_BASE_URL}${path}`;
+
+/**
+ * Manages the Todos by reading from and writing to the backend API.
+ *
+ * @returns {Object} with the following properties:
+ *   `isLoading`: a boolean indicating if the data has not yet loaded;
+ *   `todos`: a list of To Do items;
+ *   `addTodo`: a function (newTodo) that will add a new To Do;
+ *   `toggleCompleted`: a function (todoId) that will toggle the completion of an existing To Do.
+ */
+export function useTodos() {
+  const [todos, setTodos] = useState();
+  useEffect(() => {
+    /* TODO: If `todos` hasn't been defined yet, read the items from the API
+     * and set them into the `todos` state.
+     */
+    if (!todos) {
+      (async () => {
+    const jsonData = await readTodos();
+        setTodos(jsonData);
+      })();
+    }
+  }, [todos]);
+
+  function addTodo(newTodo) {
+    /* TODO: Call the API function that creates a Todo item.
+     * When the promise returned by that function resolves, update the `todos` state.
+     * Note that it is critical that you pass a _new_ array. Do not modify the `todos` array.
+     */
+    (async () => {
```

Comment:
You don't need to do the async function trick here. It's OK for `addTodo` to be an async function.